# HG changeset patch
# Parent 24f2957e878f089acc1a8fab12506c6af9f51de5

Click on company icon to edit the sign. (r24510)

diff -r 24f2957e878f src/lang/english.txt
--- a/src/lang/english.txt	Sat Sep 01 10:03:23 2012 +0200
+++ b/src/lang/english.txt	Tue Sep 04 12:58:19 2012 +0200
@@ -2749,7 +2749,7 @@
 STR_SIGN_LIST_CLEAR                                             :{BLACK}Clear filter
 STR_SIGN_LIST_MATCH_CASE_TOOLTIP                                :{BLACK}Toggle matching case when comparing sign names against the filter string
 STR_SIGN_LIST_CLEAR_TOOLTIP                                     :{BLACK}Clear filter string
-STR_SIGN_LIST_CLICK_ON_LIST_TOOLTIP                             :{BLACK}Click on sign to centre main view on sign position. Ctrl+Click opens an extra viewport centered on sign location
+STR_SIGN_LIST_CLICK_ON_LIST_TOOLTIP                             :{BLACK}Click on sign name to centre main view on sign position; click on company icon to edit the sign text. Ctrl+Click opens an extra viewport centered on sign location.
 
 # Sign window
 STR_EDIT_SIGN_CAPTION                                           :{WHITE}Edit sign text
diff -r 24f2957e878f src/signs_gui.cpp
--- a/src/signs_gui.cpp	Sat Sep 01 10:03:23 2012 +0200
+++ b/src/signs_gui.cpp	Tue Sep 04 12:58:19 2012 +0200
@@ -255,8 +255,10 @@
 				const Sign *si = this->signs[id_v];
 				if (_ctrl_pressed) {
 					ShowExtraViewPortWindow(TileVirtXY(si->x, si->y)); // Ctrl+click to open a viewport centered on the sign.
+				} else if (pt.x < this->text_offset) {
+					ShowRenameSignWindow(si);                          // Click on company icon to edit the sign.
 				} else {
-					ScrollMainWindowToTile(TileVirtXY(si->x, si->y));  // Click to move to sign position.
+					ScrollMainWindowToTile(TileVirtXY(si->x, si->y));  // Click on sign name to move to sign position.
 				}
 				break;
 			}
